File: csharp/math/Sieve.cs
// Print the primes below 100, sieved over a bool array and gathered with LINQ.
// Run: dotnet run Sieve.cs (file-based app, .NET 10 or later), or as Program.cs in a console project

const int limit = 100;

var isPrime = new bool[limit];
for (var n = 2; n < limit; n++)
{
    isPrime[n] = true;
}

for (var n = 2; n * n < limit; n++)
{
    if (!isPrime[n])
    {
        continue;
    }

    for (var multiple = n * n; multiple < limit; multiple += n)
    {
        isPrime[multiple] = false;
    }
}

Console.WriteLine(string.Join(" ", Enumerable.Range(0, limit).Where(n => isPrime[n])));

File: java/math/Sieve.java
// Print the primes below 100, sieved over a boolean array and gathered through an IntStream.
// Run: java Sieve.java

import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Sieve {
    static final int LIMIT = 100;

    public static void main(String[] args) {
        boolean[] isPrime = new boolean[LIMIT];
        for (int n = 2; n < LIMIT; n++) {
            isPrime[n] = true;
        }

        for (int n = 2; n * n < LIMIT; n++) {
            if (!isPrime[n]) {
                continue;
            }
            for (int multiple = n * n; multiple < LIMIT; multiple += n) {
                isPrime[multiple] = false;
            }
        }

        String primes = IntStream.range(0, LIMIT)
                .filter(n -> isPrime[n])
                .mapToObj(Integer::toString)
                .collect(Collectors.joining(" "));
        System.out.println(primes);
    }
}
